docs/labels.md

```diff
@@ -6,88 +6,106 @@ weight: 10
+to more easily identify resource relationships. This document describes the label propagation scheme, automatic labeling, and
+provides usage examples.
 
 ---
 
-- [Propagation Details](#propagation-details)
-- [Automatically Added Labels](#automatically-added-labels)
-- [Examples](#examples)
+- [Label propagation](#label-propagation)
+- [Automatic labeling](#automatic-labeling)
+- [Usage examples](#usage-examples)
 
 ---
 
-## Propagation Details
-
-For `Pipelines` executed using a `PipelineRun`, labels are propagated
-automatically from `Pipelines` to `PipelineRuns` to `TaskRuns` and then to
-`Pods`. Additionally, labels from the `Tasks` referenced by `TaskRuns` are
-propagated to the corresponding `TaskRuns` and then to `Pods`.
-
-For `TaskRuns` executed directly, not as part of a `Pipeline`, labels are
-propagated from the referenced `Task` (if one exists, see the
-[Specifying a `Task`](taskruns.md#specifying-a-task) section of the `TaskRun`
-documentation) to the corresponding `TaskRun` and then to the `Pod`.
-
-For `Conditions`, labels are propagated automatically to the corresponding `TaskRuns`
-and then to `Pods`.
-
-## Automatically Added Labels
-
-The following labels are added to resources automatically:
-
-- `tekton.dev/pipeline` is added to `PipelineRuns` (and propagated to `TaskRuns`
-  and `Pods`), and contains the name of the `Pipeline` that the `PipelineRun`
-  references.
-- `tekton.dev/pipelineRun` is added to `TaskRuns` (and propagated to `TaskRuns`
-  and `Pods`) that are created automatically during the execution of a
-  `PipelineRun`, and contains the name of the `PipelineRun` that triggered the
-  creation of the `TaskRun`.
-- `tekton.dev/task` is added to `TaskRuns` (and propagated to `Pods`) that
-  reference an existing `Task` (see the
-  [Specifying a `Task`](taskruns.md#specifying-a-task) section of the `TaskRun`
-  documentation), and contains the name of the `Task` that the `TaskRun`
-  references.
-- `tekton.dev/clusterTask` is added to `TaskRuns` (and propagated to `Pods`) that
-  reference an existing `ClusterTask`and contains the name of the `ClusterTask` 
-  that the `TaskRun` references. For backwards compatibility, `TaskRuns` that
-  reference a `ClusterTask` will also receive `tekton.dev/task`.
-- `tekton.dev/taskRun` is added to `Pods`, and contains the name of the
-  `TaskRun` that created the `Pod`.
-- `app.kubernetes.io/instance` and `app.kubernetes.io/component` is added to 
-  Affinity Assistant `StatefulSets` and `Pods`. These are used for Pod Affinity for TaskRuns.
-
-## Examples
-
-- [Finding Pods for a Specific PipelineRun](#finding-pods-for-a-specific-pipelinerun)
-- [Finding TaskRuns for a Specific Task](#finding-taskruns-for-a-specific-task)
-
-### Finding Pods for a Specific PipelineRun
-
-To find all `Pods` created by a `PipelineRun` named test-pipelinerun, you could
-use the following command:
+## Label propagation
```

Comment:
One general question that might be good to answer here:

What happens if a label already exists? Do we stomp over it?

I think we preserve the existing one, but would have to confirm.

Reply:
I'll be happy to add a clarification. Can someone verify this?

Reply:
> What happens if a label already exists? Do we stomp over it?

we overwrite existing labels 😢 see open question in issue #2934

> E.g. For standalone `TaskRuns` (that is, ones not executing as part of a `Pipeline`), labels
propagate from the [referenced `Task`](taskruns.md#specifying-the-target-task), if one exists, to
the corresponding `TaskRun`, and then to the associated `Pod`. 

Labels from referenced `Task` propagate to `TaskRun` and overwrites if same labels exist in `TaskRun`.

I seems bigger discussion and worth bringing it to API Working group 😸

Reply:
Thanks! Let's make another PR after the discussion takes place and merge this PR in the meantime.

Reply:
We will be discussing this in API WG on 7/20 😄
yup @sergetron we can create separate PR after the discussion.

Reply:
Excellent, resolving for now!